src/amo/sagas/reviews.js

```diff
@@ -108,6 +113,42 @@ function* fetchReviews({
   }
 }
 
+function* fetchReviewPermissions({
+  payload: { errorHandlerId, addonId, userId },
+}: FetchReviewPermissionsAction): Generator<any, any, any> {
+  const errorHandler = createErrorHandler(errorHandlerId);
+
+  try {
+    yield put(errorHandler.createClearingAction());
+    const state: AppState = yield select(getState);
+
+    const params: GetReviewsParams = {
+      addon: addonId,
+      apiState: state.api,
+      show_permissions_for: userId,
+    };
+
+    const response: GetReviewsApiResponse = yield call(getReviews, params);
+    invariant(
+      response.can_reply !== undefined,
+      'response.can_reply was unexpectedly undefined',
+    );
```

Comment:
> Why is that handled at this level?

Since it's an optional response property, Flow wants the code to be defensive against undefined values. Also, I think it will be helpful to throw an error here to make it unambiguous that the API returned a broken response (which shouldn't happen).

<details>
<summary>Flow's complaint</summary>

```
Cannot call setReviewPermissions with object literal bound to the first parameter because undefined [1] is incompatible
with boolean [2] in property canReplyToReviews.

     src/amo/sagas/reviews.js
     135|     //);
     136|
     137|     yield put(
     138|       setReviewPermissions({
     139|         addonId,
     140|         canReplyToReviews: response.can_reply,
     141|         userId,
     142|       }),
     143|     );
     144|   } catch (error) {
     145|     log.warn(

     src/amo/actions/reviews.js
 [2] 225|   canReplyToReviews: boolean,

     src/amo/api/reviews.js
 [1] 163|   can_reply?: boolean,
```

</details>

> It seems this check already exists in setReviewPermissions().

Yes but that's really only for unit tests or anything that doesn't have Flow coverage. If something else calls `setReviewPermissions()` without defining this parameter, it means the caller made a mistake. That call may or may not happen with the result of an API response.